server/graphql/v2/input/TransactionReferenceInput.ts

```diff
@@ -18,27 +17,20 @@ const TransactionReferenceInput = new GraphQLInputObjectType({
   }),
 });
 
-const getDatabaseIdFromTransactionReference = (input: Record<string, unknown>): number => {
-  if (input['id']) {
-    return idDecode(input['id'], IDENTIFIER_TYPES.TRANSACTION);
-  } else if (input['legacyId']) {
-    return <number>input['legacyId'];
-  } else {
-    return null;
-  }
-};
-
 /**
  * Retrieve an expense from an `ExpenseReferenceInput`
  */
 const fetchTransactionWithReference = async (
   input: Record<string, unknown>,
   { loaders = null, throwIfMissing = false } = {},
 ): Promise<typeof models.Transaction> => {
-  const dbId = getDatabaseIdFromTransactionReference(input);
   let transaction = null;
-  if (dbId) {
-    transaction = await (loaders ? loaders.Transaction.byId.load(dbId) : models.Transaction.findByPk(dbId));
+  if (input.id) {
+    transaction = await models.Transaction.findOne({ where: { uuid: input.id } });
```

Comment:
On the Frontend, this is going to have an impact. People who trigger mutations from an already-loaded page while after we deploy the change my face errors like `Can't refund ... transaction does not exists`. We can either accept that as a small-enough risk (easily solved by reloading the page), or we implement a fallback like:

```es6
transaction = await models.Transaction.findOne({ where: { uuid: input.id } });
if (!transaction) {
  try {
    id = idDecode(input.id, IDENTIFIER_TYPES.TRANSACTION);
    transaction = await models.Transaction.findByPk(input.id});
  } catch {
    // Ignore invalid GQLV2 IDs
  }
}
```

Reply:
I'll take the risk and remove `IDENTIFIER_TYPES.TRANSACTION`